### ISSUE TITLE
Easier squash

### ISSUE THREAD

**Issue by francoas** [gitextensions/gitextensions#5608]
**Do you want to request a *feature* or report a *bug*?**

feature

**What is the expected behavior?**

I'd like an easier U/X way to create squash commit like gitKraken does

https://www.youtube.com/watch?v=cr1N8VTRmfM

That is: select the commits to squash and squash them

**Comment by vbjay** [comment]
Do an interactive rebase.  It's a better way that lets you verify what commits will be affected.

**Comment by vbjay** [comment]
We do have the advanced menu that creates a squash or fixup commit. If you
are squashing existing commits just rebase interactively.

**Comment by vbjay** [comment]
FYI Dark theme: #5489

**Comment by RussKie** [comment]
Agree. Let's get v3 out of the door, and work further on refining the UX.

**Comment by NikolayXHD** [comment]
meanwhile the feature is being not implemented, the way I do commit squashing with gitextensions:
https://stackoverflow.com/a/47464064/6656775

**Comment by RussKie** [comment]
Right click on the oldest parent, select reset current branch to here,
perform soft reset, commit.

**Comment by pmiossec** [comment]
> Right click on the oldest parent, select reset current branch to here, perform soft reset, commit.

I don't know if this workflow could be improve a little by adding 2 options:
* stashing current working directory and index changes because it could interfere with the "squash" workflow
* allowing the user to have the commit message pre-filled with the concatenation of all the commit messages of the commits squashed

**Comment by RussKie** [comment]
Anything is possible, but there are many moving parts and decision points
that I'd rather leave to a user.

In the linked video (gitkraken iirc) they explained how many constraints
they made. Implementing those wouldn't be too difficult, but providing a
good UX - would definitely be a challenge.

Overall, the value-add is unclear to me.

**Comment by vbjay** [comment]
Rebase interactive. Adjust Todo to do as many squashes or fixups and complete rebase.

**Comment by francoas** [comment]
> Anything is possible, but there are many moving parts and decision points that I'd rather leave to a user. In the linked video (gitkraken iirc) they explained how many constraints they made. Implementing those wouldn't be too difficult, but providing a good UX - would definitely be a challenge. Overall, the value-add is unclear to me.

Well, most constraints they showed are banally obvious (like "more than one commit selected" and "genealogical consecutive") and even enabling the squash command only on the condidition "youngest commit is the head" etc...  the situation is the most common to me: I develop a new feature in a different branch. I commit and push often for backup reason more than for feature identification, Prior to  merging I want to squash. 

ACTUALLY WHAT THEY DO is simply a  
- Right click on the oldest parent, 
- select reset current branch to here,
- perform soft reset, 
- commit.

(Please avoid to explain it, we all know)

But this is the point. We know and we can do this because we are advanced user (as we all are deveolopers)
But believe me: it is very difficult to teach that to novices or to people that are not developers but only configurators  and they often are afraid to make a mistake. THEY ARE AFRAID OF THE "RESET" COMMAND, 

So a squash command DOESN'T INTRODUCE NOTHING THAT CAN'T BE DONE OTHERWISE but is just an improvement in the user experience, 

When in a previous comment I said "I have in my team more and more people abandoning git extension for git kraken just for its easy-to-use interface and U/X" I was not lying or exaggerating.  
It's just a metter of forma mentis. Provide only what is necessary or what is liked ? 
That's the same difference between an I/OS and Android phone. Some people love Iphone non because it offers better functionalities but because it's easier to use.

SO THIS IS THE POINT OF THIS ISSUE: 

NOT TO DISCUSS IF IT IS NECESSARRY 
OR HOW IT CAN BE DONE OTHERWISE (this is really stupid unless it can be done otherwise with JUST ONE CLICK)

but if there is the WILL to give a better U/X or not.
And if one thing can be done with just one click less for me is enough to improve it (at least in the I/F we develop)

Otherwise le'ts go back to the shell commands.

I beg your pardon for the outburst, but I am tired of replies explaining how it can be done otherwise. 
Once for all: We all already know how to do it otherwise:

https://github.com/gitextensions/gitextensions/wiki/How-To%3A-Squash-and-Rebase-your-changes#squash

The issue was not "there is no way to squash commits" but "I'd like a simple squash command like the GitKracken one"

Cheers

**Comment by mstv** [comment]
> but if there is the WILL to give a better U/X or not.

Feel free to contribute. PRs are welcome, (but SHouting is not).

**Comment by francoas** [comment]
> > but if there is the WILL to give a better U/X or not.
> 
> Feel free to contribute. PRs are welcome, (but SHouting is not).

Not shouting at all (capital letters can be also used to emphasize a concept) but expressing my frustration for replies that are completely inconsistent with the subject. 
its like proposing to install a tap in a bathroom instead to go to the river to take water and receiving answers like “if you need water you can go to the river”

**Comment by vbjay** [comment]
Squash and fixup commits don't do what you think. With the autosquash feature of git turned on it helps you auto edit the to-do list for an interactive rebase.

**Comment by RussKie** [comment]
Closing as stale. If someone is keen on implementing this and sending a pull-request, we'll certainly consider it.
